mysql-test/suite/period/t/overlaps.test

```diff
@@ -0,0 +1,139 @@
+                          period for p(s,e),
+                          primary key(id, q without overlaps));
+
+--error ER_PERIOD_NOT_FOUND
+create or replace table t(id int, s date, e date,
+                          primary key(id, p without overlaps));
+
+--error ER_KEY_CONTAINS_PERIOD_FIELDS
+create or replace table t(id int, s date, e date,
+                          period for p(s,e),
+                          primary key(id, s, p without overlaps));
+
+create or replace table t(id int, s date, e date,
+                          period for p(s,e),
+                          primary key(id));
+insert into t values (1, '2003-03-01', '2003-05-01');
+--error ER_DUP_ENTRY
+insert into t values (1, '2003-04-01', '2003-05-01');
+
+create or replace table t(id int, u int, s date, e date,
+                          period for p(s,e),
+                          primary key(id, p without overlaps),
+                          unique(u));
+show create table t;
+insert into t values (1, 1, '2003-03-01', '2003-05-01');
+insert into t values (1, 2, '2003-05-01', '2003-07-01');
+--error ER_DUP_KEY
+insert into t values (1, 3, '2003-04-01', '2003-05-01');
+
+create or replace table t(id int, u int, s date, e date,
+                          period for p(s,e),
+                          primary key(id, p without overlaps),
+                          unique(u, p without overlaps));
+show create table t;
+insert into t values (1, 1, '2003-03-01', '2003-05-01');
+insert into t values (1, 2, '2003-05-01', '2003-07-01');
+insert into t values (2, 1, '2003-05-01', '2003-07-01');
+
+create or replace table t(id int, s date, e date,
+                          period for p(s,e)) engine=innodb;
+
+insert into t values (1, '2003-01-01', '2003-03-01'),
+                     (1, '2003-05-01', '2003-07-01'),
+                     (1, '2003-02-01', '2003-04-01');
+
+--replace_regex /#sql-\w+/#sql-temp/
+--error ER_DUP_KEY
+alter table t add primary key(id, p without overlaps);
+
+--echo # Historical rows are not checked against constraints
+set @@system_versioning_alter_history= keep;
+alter table t add system versioning;
+delete from t;
+alter table t add primary key(id, p without overlaps);
+
+insert into t values (1, '2003-01-01', '2003-03-01'),
+                     (1, '2003-03-01', '2003-05-01');
+
+--error ER_PERIOD_WITHOUT_OVERLAPS_PARTITIONED
+create or replace table t1 (x int, s date, e date,
+                            period for p(s,e),
+                            primary key(x, p without overlaps)
+) with system versioning
+  partition by key (x);
+
+--error ER_PERIOD_WITHOUT_OVERLAPS_PARTITIONED
+create or replace table t1 (x int, s date, e date,
+                            period for p(s,e),
+                            primary key(x, p without overlaps)
+) with system versioning
+  partition by system_time limit 2
+  (partition p0 history,
+   partition p1 history,
+   partition pn current);
+
+create or replace database test;
```

Comment:
## Bug 5: ON DUPLICATE KEY doesn't work
### Reproduce
```sql
create or replace table t(id int, s date, e date,
                          period for p(s,e),
                          primary key(id, p without overlaps));
insert into t values (1, '2003-03-01', '2003-05-01');
insert into t values (1, '2003-03-01', '2003-05-01') on duplicate key update id= id + 100;
```
### Result
```
ERROR 1022 (23000): Can't write; duplicate key in table 't'
```
### Expected
No error.

Reply:
That's also a subject of MDEV-17395. Need to rename the ticket, though, because no 'FOR PROTION' syntax is going to be introduced there

Reply:
Ok. Postponed.